ui/components-react/src/components-react/newEditors/values/Metadata.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Value types supported by editor system.
+ * @beta
+ */
+export type ValueType =
+  | "string"
+  | "number"
+  | "bool"
+  | "date"
+  | "enum"
+  | "instanceKey";
+
+/**
+ * Additional metadata that is used along side value to determine applicable editor.
+ * @beta
+ */
+export interface ValueMetadata {
+  type: ValueType;
+  preferredEditor?: string;
+}
+
+/**
+ * Type definition for available enum choice that can be supplied for editing enum values.
+ */
+export interface EnumChoice {
+  value: number | string;
+  label: string;
+}
+
+/**
+ * Additional metadata that is used along side enum value to determine applicable editor.
+ */
+export interface EnumValueMetadata extends ValueMetadata {
```

Comment:
I think this type should also have `isStrict: boolean`. See `isStrict` here: https://www.itwinjs.org/reference/ecschema-metadata/metadata/enumeration/

Reply:
Added `isStrict`.